File: frontend/src/api/quota.ts
import { apiFetch } from './client'

export interface QuotaSummaryItem {
  project_id: number
  project_name: string
  daily_limit: number
  units_used: number
  remaining: number
  percentage_used: number
  videos_today: number
  video_limit: number
}

export interface QuotaSummaryResponse {
  projects: QuotaSummaryItem[]
}

export interface QuotaEstimateResponse {
  estimated_cost: number
  remaining_quota: number
  sufficient: boolean
}

export async function getQuotaSummary(): Promise<QuotaSummaryResponse> {
  return apiFetch<QuotaSummaryResponse>('/quota/summary')
}

export async function getQuotaEstimate(
  numVideos: number,
  hasThumbnail: boolean,
): Promise<QuotaEstimateResponse> {
  return apiFetch<QuotaEstimateResponse>('/quota/estimate', {
    method: 'POST',
    body: JSON.stringify({ file_count: numVideos, has_thumbnail: hasThumbnail }),
  })
}

File: backend/schemas/quota.py
from pydantic import BaseModel


class QuotaEstimateRequest(BaseModel):
    file_count: int
    has_thumbnail: bool = False


class QuotaEstimateResponse(BaseModel):
    estimated_cost: int
    remaining_quota: int
    sufficient: bool


class QuotaSummaryItem(BaseModel):
    project_id: int
    project_name: str
    daily_limit: int
    units_used: int
    remaining: int
    percentage_used: float
    videos_today: int
    video_limit: int


class QuotaSummaryResponse(BaseModel):
    projects: list[QuotaSummaryItem]
